daemon/sdnotify_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+)
+
+// TestSdNotify
+func TestSdNotify(t *testing.T) {
+	testDir, err := ioutil.TempDir("/tmp/", "test-")
+	defer os.RemoveAll(testDir)
+
+	notifySocket := testDir + "/notify-socket.sock"
+	laddr := net.UnixAddr{
+		Name: notifySocket,
+		Net:  "unixgram",
+	}
+	_, e := net.ListenUnixgram("unixgram", &laddr)
+	if e != nil {
+		t.Errorf("Error creating the notification socket")
+		panic(e)
```

Comment:
panic tends to be used to indicate failure during test set-up that is distinct from the behaviour being tested